x/storage/keeper/keeper.go

```diff
@@ -341,6 +341,15 @@ func (k Keeper) ForceDeleteBucket(ctx sdk.Context, bucketId sdkmath.Uint, cap ui
 				return false, deleted, err
 			}
 			k.SetInternalBucketInfo(ctx, bucketInfo.Id, internalBucketInfo)
+
+			// if an object is updating, also need to unlock the shadowObject fee
+			if objectInfo.IsUpdating {
```

Comment:
Isn't this a hardfork?

Reply:
Because this feature not eanbled on testnet I think, so it is not hardfork?